render/respond.go

```diff
@@ -0,0 +1,111 @@
+package render
+
+import (
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"net/http"
+	"reflect"
+
+	"golang.org/x/net/context"
+)
+
+var Respond = DefaultRespond
+
+func DefaultRespond(ctx context.Context, w http.ResponseWriter, v interface{}) {
```

Comment:
Responder should be a middleware and get it's body and headers from context.

This is go 1.7 middleware with context:
```go
func someMiddleware(next http.Handler) http.Handler {
    return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
        // before code
        next.ServeHTTP(w, r)
        // after code
    })
}
```
By having responder write to response writer in handler you prevent all middlewares in chain to set HTTP headers in `// after code`
IMO having `"response_headers": *http.Header` and `"response_body": interface{}` in context + responder as a middleware mounted very high (top level by default?) up the chain would be more flexible.